home/models.py

```diff
@@ -464,6 +467,19 @@ class Meta:
         indexes = [
             Index(fields=["status"])
         ]
+    
+    def get_content_with_course_links(self):
+        content = escape(self.content)
+        course_code_format = '^([A-Z]{4}(?:[A-Z]|[0-9]){3,6})$'
+        courses_replaced = []
+
+        for original_word in re.split(r' |\.', content):
```

Comment:
Both good ideas. I will implement them both. 

Do you also want me to implement this in course.py line 78-85 where we generate links on course descriptions?

Reply:
yup, that would be great!

Reply:
@kandasamyc I have added a commit so it only iterates over matches instead of every word.

After reviewing it more, I don't think we should only match full words.

1.	There are many valid cases that adding ( |\\.) to the start and end will miss. Below are a few: 
     a.	I loved MATH140H!
     b.	The professor said, "MATH140H is one of the hardest courses we teach"
     c.	 My tip for MATH140H: don't take it.
     d.	I really enjoyed calc 1 (MATH140H).
     e.	Why would anyone choose to take MATH140H?
2.	I can't think of many instances where there would be a false positive string that isn’t a course code, so the speed up would likely be small.
3.    The performance gain comes at the cost of accuracy which is only worth it if there are pages loading slow specifically because of this logic. I am not seeing any slow down on review pages locally, not sure how production will fare though.

That being said, If you feel strongly about this let me know and I'll change it to match only full words and try to account for all the end punctuation I can.

Reply:
That's a fair point - works for me, thanks for the changes!